src/sparse_merkle_tree/smt.rs

```diff
@@ -1394,6 +1397,11 @@ impl SparseMerkleTree {
             } else if !query.binary_bitmap[0] {
                 sibling_hash = Some(EMPTY_HASH.to_vec());
             } else if query.binary_bitmap[0] {
+                if sibling_hashes.len() == next_sibling_hash {
+                    return Err(SMTError::InvalidInput(String::from(
+                        "length of sibnling_hashes does not match with bitmap",
```

Comment:
You have a typo in the error string. Also, I'm not sure if the error is describing the issue very well, as the length of sibling hashes does not have to do with the bitmap only but also with the keys you are querying. I would write something like, "no more sibling hashes available" or something like that.